runtime/caml/tsan.h

```diff
@@ -22,8 +22,16 @@
 #if defined(__has_feature)
 #  if __has_feature(thread_sanitizer)
 #    undef CAMLreally_no_tsan
-#    define CAMLreally_no_tsan \
-       __attribute__((disable_sanitizer_instrumentation))
+#    if defined(__has_attribute)
+#      if __has_attribute(disable_sanitizer_instrumentation)
+#        define CAMLreally_no_tsan \
+           __attribute__((disable_sanitizer_instrumentation))
+#      else
+#        define CAMLreally_no_tsan __attribute__((no_sanitize("thread")))
+#      endif
+#    else
+#      define CAMLreally_no_tsan __attribute__((no_sanitize("thread")))
+#    endif
```

Comment:
Could we rephrase as follows?

```c
#if defined(__has_attribute) && has_attribute(disable_sanitizer_instrumentation)
#  define CAMLreally_no_tsan __attribute__((disable_sanitizer_instrumentation))
#else
#  define CAMLreally_no_tsan __attribute__((no_sanitize("thread")))
```

Reply:
I suspect ASAN and TSAN are mutually incompatible, so this discussion may be moot.  But, at the risk of beating a dead horse, why is `disable_sanitizer_instrumentation` preferable (if it is available) to `no_sanitize("thread")` ?

Reply:
> Could we rephrase as follows?

I asked myself the same question but had a doubt whether the `&&` operator is lazy in CPP. If it’s not, we could get errors.

Reply:
Indeed, testing indicates that the following raises a warning

```
#if defined(foo) && foo("bar")
"foo"
#endif
```

(This works fine for non-function-like macros.)